### PR TITLE
Add `schema_locked` storage parameter for changefeed performance improvement

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-10681, DOC-10680

PR adds the `schema_locked` storage parameter to the table storage param page and an example to the `create changefeed` page.

This was previously not documented, but adding docs here through to v23.1.

### CODE REVIEW

**Comment by kathancox** [comment]
cc @rharding6373

**Comment by kathancox** [comment]
Hi @andyyang890, just a friendly ping on a review for this PR!

File: src/current/v23.1/create-changefeed.md [hunk 1]
```diff
@@ -335,6 +335,19 @@ CREATE CHANGEFEED FOR TABLE table_name INTO 'external://kafka_sink'
...
+
+Enable `schema_locked` on the watched table with the [`ALTER TABLE`]({% link {{ page.version.version }}/alter-table.md %}) statement:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER TABLE watched_table SET (schema_locked = true);
+~~~
+
+While `schema_locked` is enabled on a table, attempted schema changes on the table will be rejected and an error returned. If you need to run a schema change on the locked table, unlock the table with `schema_locked = false`, complete the schema change, and then lock the table again with `schema_locked = true`. The changefeed will run as normal while `schema_locked = false`, but it will not benefit from the performance optimization.
```
Comment (andyyang890) [inline src/current/v23.1/create-changefeed.md:349]: Maybe we should also include some example syntax for how to reset the field. They can do that by doing either:
```
ALTER TABLE watched_table SET (schema_locked = false);
```
or
```
ALTER TABLE watched_table RESET (schema_locked);
```

Ditto for all the other versions.

**Review by andyyang890 — APPROVED** [review]
Looks good to me! Just had one suggestion.

File: src/current/v23.1/create-changefeed.md (hunk 1 above)
Comment (kathancox) [reply src/current/v23.1/create-changefeed.md:349]: Done!

File: src/current/_includes/v23.1/misc/table-storage-parameters.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 |                    Parameter name                    |                                                                    Description                                                                    | Data type | Default value |
 |------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|-----------|---------------|
 | `exclude_data_from_backup`                           | Exclude the data in this table from any future backups.                                                                                           | Boolean   | `false`       |
+| <span class="version-tag">New in v23.1.15:</span> `schema_locked` | Disallow schema changes on this table. Enabling `schema_locked` can help [improve performance of changefeeds]({% link {{ page.version.version }}/create-changefeed.md %}#disallow-schema-changes-on-tables-to-improve-changefeed-performance) running on this table. | Boolean | `false` |
```
Comment (rmloveland) [inline src/current/_includes/v23.1/misc/table-storage-parameters.md:4]: the mention of 'schema changes' could link to schema changes docs for this version eg https://www.cockroachlabs.com/docs/v23.1/online-schema-changes

File: src/current/_includes/v23.2/misc/table-storage-parameters.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 |                    Parameter name                    |                                                                    Description                                                                    | Data type | Default value |
 |------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|-----------|---------------|
 | `exclude_data_from_backup`                           | Exclude the data in this table from any future backups.                                                                                           | Boolean   | `false`       |
+| <span class="version-tag">New in v23.2.1:</span> `schema_locked` | Disallow schema changes on this table. Enabling `schema_locked` can help [improve performance of changefeeds]({% link {{ page.version.version }}/create-changefeed.md %}#disallow-schema-changes-on-tables-to-improve-changefeed-performance) running on this table. | Boolean | `false` |
```
Comment (rmloveland) [inline src/current/_includes/v23.2/misc/table-storage-parameters.md:4]: same re: could link to schema change docs

File: src/current/_includes/v24.1/misc/table-storage-parameters.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 |                    Parameter name                    |                                                                    Description                                                                    | Data type | Default value |
 |------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|-----------|---------------|
 | `exclude_data_from_backup`                           | Exclude the data in this table from any future backups.                                                                                           | Boolean   | `false`       |
+| `schema_locked` | Disallow schema changes on this table. Enabling `schema_locked` can help [improve performance of changefeeds]({% link {{ page.version.version }}/create-changefeed.md %}#disallow-schema-changes-on-tables-to-improve-changefeed-performance) running on this table. | Boolean | `false` |
```
Comment (rmloveland) [inline src/current/_includes/v24.1/misc/table-storage-parameters.md:4]: same re: could link to schema change docs

File: src/current/_includes/v24.2/misc/table-storage-parameters.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 |                    Parameter name                    |                                                                    Description                                                                    | Data type | Default value |
 |------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|-----------|---------------|
 | `exclude_data_from_backup`                           | Exclude the data in this table from any future backups.                                                                                           | Boolean   | `false`       |
+| `schema_locked` | Disallow schema changes on this table. Enabling `schema_locked` can help [improve performance of changefeeds]({% link {{ page.version.version }}/create-changefeed.md %}#disallow-schema-changes-on-tables-to-improve-changefeed-performance) running on this table. | Boolean | `false` |
```
Comment (rmloveland) [inline src/current/_includes/v24.2/misc/table-storage-parameters.md:4]: same re: could link to schema change docs

**Review by rmloveland — APPROVED** [review]
LGTM, had some annoying link suggestions!

**Review by andyyang890 — APPROVED** [review]
New changes LGTM!

**Pull request merged**